### PR TITLE
Kamelet: Ensure camel route input is not null.

### PR DESCRIPTION
**Pull request by jameshilliard**
Fixes `NullPointerException`

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by davsclaus** [comment]
I wonder what example / use-case you had that could cause this NPE? I am asking because a camel  route should always have 1 input.

**Comment by jameshilliard** [comment]
> I wonder what example / use-case you had that could cause this NPE?

I was messing around with route templates and hit this error, here's the reproducer:
```java
diff --git a/components/camel-kamelet/src/test/java/org/apache/camel/component/kamelet/KameletRouteTest.java b/components/camel-kamelet/src/test/java/org/apache/camel/component/kamelet/KameletRouteTest.java
index 15c0cc853fa..43a02038760 100644
--- a/components/camel-kamelet/src/test/java/org/apache/camel/component/kamelet/KameletRouteTest.java
+++ b/components/camel-kamelet/src/test/java/org/apache/camel/component/kamelet/KameletRouteTest.java
@@ -56,6 +56,7 @@ public class KameletRouteTest extends CamelTestSupport {
             public void configure() throws Exception {
                 routeTemplate("echo")
                         .templateParameter("prefix")
+                        .asRouteDefinition()
                         .from("kamelet:source")
                         .setBody().simple("{{prefix}}-${body}");
```

**Comment by davsclaus** [comment]
Okay but as an end user you should not call `.asRouteDefinition()` its for internal use

**Comment by davsclaus** [comment]
Instead of avoiding the NPE then you can update the PR to check that input is not null and report an error if so.

**Comment by jameshilliard** [comment]
> Okay but as an end user you should not call `.asRouteDefinition()` its for internal use

hmm, shouldn't it not be public if that is the case?

**Comment by davsclaus** [comment]
The method is used from other parts in camel that are not in the same package, so it has to be public.

File: components/camel-kamelet/src/main/java/org/apache/camel/component/kamelet/Kamelet.java [hunk 1]
```diff
@@ -116,6 +116,10 @@ public static RouteDefinition templateToRoute(RouteTemplateDefinition in, Map<St
         RouteDefinition def = in.asRouteDefinition();
         def.setId(rid);
 
+        if (def.getInput() == null) {
+            throw new IllegalArgumentException("Camel route " + rid + " input does not exist.");
```
Comment (jameshilliard) [inline components/camel-kamelet/src/main/java/org/apache/camel/component/kamelet/Kamelet.java:120]: How's this wording for the error message?

Comment (davsclaus) [reply components/camel-kamelet/src/main/java/org/apache/camel/component/kamelet/Kamelet.java:120]: Yes this is fine

**Review by davsclaus — APPROVED** [review]

**Pull request merged**